### PR TITLE
feat(galaxy-ansible): Switching from v2 api version to v3

### PR DESCRIPTION
**Pull request by lmilbaum**
## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/gomod/artifacts.ts
Comment (rarkins) [inline lib/modules/manager/gomod/artifacts.ts:1]: Revert all changes in this file

**Review by rarkins — CHANGES_REQUESTED** [review]

File: lib/modules/manager/gomod/artifacts.ts
Comment (lmilbaum) [reply lib/modules/manager/gomod/artifacts.ts:1]: I pushed it by mistake. Removed

**Review by rarkins — CHANGES_REQUESTED** [review]
Please test against a public/demo repo to verify it all works as expected

File: lib/modules/datasource/galaxy-collection/index.ts [hunk 1]
```diff
@@ -36,7 +36,7 @@ export class GalaxyCollectionDatasource extends Datasource {
     const [namespace, projectName] = packageName.split('.');
 
     // TODO: types (#22198)
-    const baseUrl = `${registryUrl}api/v2/collections/${namespace}/${projectName}/`;
+    const baseUrl = `${registryUrl}api/v3/collections/${namespace}/${projectName}/`;
```
Comment (HonkingGoose) [inline lib/modules/datasource/galaxy-collection/index.ts:39]: Are you sure this is a `fix` type PR? Switching to a new API _may_ be a breaking change.

I guess at the very least this kind of change should go in a `feat:` type PR, so it's not mingled with "real patches"?

Comment (secustor) [reply lib/modules/datasource/galaxy-collection/index.ts:39]: I agree this should be a `feat`.

Also is v3 really returning the exact same response on the same sub path?

Comment (lmilbaum) [reply lib/modules/datasource/galaxy-collection/index.ts:39]: Thank you for your feedback. Will change it to feat.
@secustor I've started to check it and the response is probably different. I don't know typescript enough to understand the errors it throws.
In any case, the v2 version is broken.

**Comment by lmilbaum** [comment]
The following error is raised. No idea how to start debugging it. Would appreciate some help.

```
DEBUG: Datasource unknown error (repository=platform-engineering-org/iac)
       "datasource": "galaxy-collection",
       "packageName": "amazon.aws",
       "err": {
         "message": "Cannot read properties of undefined (reading 'map')",
         "stack": "TypeError: Cannot read properties of undefined (reading 'map')\n    at GalaxyCollectionDatasource.getReleases (/workspaces/renovate/lib/modules/datasource/galaxy-collection/index.ts:71:57)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async /workspaces/renovate/lib/util/cache/package/decorator.ts:119:18\n    at async getRegistryReleases (/workspaces/renovate/lib/modules/datasource/index.ts:77:15)\n    at async fetchReleases (/workspaces/renovate/lib/modules/datasource/index.ts:289:15)\n    at async lookupUpdates (/workspaces/renovate/lib/workers/repository/process/lookup/index.ts:115:56)\n    at async withLookupStats (/workspaces/renovate/lib/workers/repository/process/fetch.ts:27:18)\n    at async fetchDepUpdates (/workspaces/renovate/lib/workers/repository/process/fetch.ts:72:30)\n    at async /workspaces/renovate/node_modules/.pnpm/p-map@4.0.0/node_modules/p-map/index.js:57:22"
       }
```

**Comment by rarkins** [comment]
I think it's far more complicated than search/replace v2 to v3, we need to plan it in an issue first

**Comment by rarkins** [comment]
Created feature request: #24981

**Pull request closed without merging**